<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BNBLEditorClass</class>
 <widget class="QMainWindow" name="BNBLEditorClass">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>536</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BNBL Editor</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QFrame" name="playground_frame">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>50</y>
      <width>256</width>
      <height>192</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="lineWidth">
     <number>1</number>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="sizeConstraint">
      <enum>QLayout::SetDefaultConstraint</enum>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QWidget" name="playground" native="true">
       <property name="styleSheet">
        <string notr="true">QWidget#playground {
	background-color: rgb(218, 218, 218);
}</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>221</width>
      <height>221</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="1" column="0">
      <widget class="QLabel" name="label_2">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Selected Rectagle:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QSpinBox" name="rectCount_sb">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="maximum">
        <number>65535</number>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Rectangle Count:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QSpinBox" name="selRect_sb">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="maximum">
        <number>65535</number>
       </property>
      </widget>
     </item>
     <item row="2" column="0" colspan="2">
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>Properties</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="2" column="0">
         <widget class="QLabel" name="label_5">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Width:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_3">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>X Position:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_4">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Y Position:</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_6">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Height:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="xPos_sb">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="maximum">
           <number>4095</number>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="yPos_sb">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="maximum">
           <number>4095</number>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="width_sb">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="maximum">
           <number>255</number>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QSpinBox" name="height_sb">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="maximum">
           <number>255</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>20</y>
      <width>241</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>2D Viewport:</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>536</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionNew"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSave_as"/>
    <addaction name="separator"/>
    <addaction name="actionClose"/>
   </widget>
   <widget class="QMenu" name="menuExtra">
    <property name="title">
     <string>2D Viewport</string>
    </property>
    <addaction name="actionOpenBg"/>
    <addaction name="actionCloseBg"/>
   </widget>
   <widget class="QMenu" name="menuFilename">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="title">
     <string/>
    </property>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuExtra"/>
   <addaction name="menuFilename"/>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="sizeGripEnabled">
    <bool>false</bool>
   </property>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionSave_as">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save as...</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Close</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionOpenBg">
   <property name="text">
    <string>Open Background Image</string>
   </property>
  </action>
  <action name="actionCloseBg">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Close Background Image</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>BNBLEditorClass</receiver>
   <slot>openFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClose</sender>
   <signal>triggered()</signal>
   <receiver>BNBLEditorClass</receiver>
   <slot>closeFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>BNBLEditorClass</receiver>
   <slot>saveFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_as</sender>
   <signal>triggered()</signal>
   <receiver>BNBLEditorClass</receiver>
   <slot>saveFileAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew</sender>
   <signal>triggered()</signal>
   <receiver>BNBLEditorClass</receiver>
   <slot>newFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpenBg</sender>
   <signal>triggered()</signal>
   <receiver>BNBLEditorClass</receiver>
   <slot>openOrCloseBg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rectCount_sb</sender>
   <signal>valueChanged(int)</signal>
   <receiver>BNBLEditorClass</receiver>
   <slot>rectCount_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>208</x>
     <y>57</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>selRect_sb</sender>
   <signal>valueChanged(int)</signal>
   <receiver>BNBLEditorClass</receiver>
   <slot>selRect_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>208</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>xPos_sb</sender>
   <signal>valueChanged(int)</signal>
   <receiver>BNBLEditorClass</receiver>
   <slot>rectProperties_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>201</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>yPos_sb</sender>
   <signal>valueChanged(int)</signal>
   <receiver>BNBLEditorClass</receiver>
   <slot>rectProperties_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>201</x>
     <y>167</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>width_sb</sender>
   <signal>valueChanged(int)</signal>
   <receiver>BNBLEditorClass</receiver>
   <slot>rectProperties_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>201</x>
     <y>207</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>height_sb</sender>
   <signal>valueChanged(int)</signal>
   <receiver>BNBLEditorClass</receiver>
   <slot>rectProperties_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>201</x>
     <y>239</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCloseBg</sender>
   <signal>triggered()</signal>
   <receiver>BNBLEditorClass</receiver>
   <slot>openOrCloseBg()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>267</x>
     <y>149</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>openFile()</slot>
  <slot>closeFile()</slot>
  <slot>saveFile()</slot>
  <slot>saveFileAs()</slot>
  <slot>newFile()</slot>
  <slot>openOrCloseBg()</slot>
  <slot>rectCount_changed(int)</slot>
  <slot>selRect_changed(int)</slot>
  <slot>rectProperties_changed()</slot>
 </slots>
</ui>
